sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -231,13 +231,17 @@ public HttpResponse logResponse(ClientLogger logger, HttpResponse response, Dura
                 long contentLength = getContentLength(logger, response.getHeaders());
 
                 if (shouldBodyBeLogged(contentTypeHeader, contentLength)) {
-                    return new LoggingHttpResponse(response, logBuilder, logger,
-                        (int) contentLength, contentTypeHeader);
+                    if (!response.isBuffered()) {
+                        response = response.buffer();
```

Comment:
would it be the case here:
- response is a network stream and client implementation does `BinaryData.fromStream(response)`
- Logger does `response.buffer()` and `response.getBody` resulting  in
  - a copy of BD
  - reading full network stream 
- When use starts reading the original response, there is nothing there

Seems when we buffer, we actually need to make the original BD replayable instead of creating a copy.

Reply:
If this were the case, we have a bug deeper than the `HttpLoggingPolicy` in either the `HttpResponse` or `BinaryData` implementation. For this, we should add tests where the request and response bodies are logged and validate the request and response body after the logging to make sure this is properly making the body replayable.